packages/target-ethers-v5/src/codegen/index.ts

```diff
@@ -0,0 +1,268 @@
+import { values } from 'lodash'
+import {
+  AbiParameter,
+  BytecodeWithLinkReferences,
+  Contract,
+  EventArgDeclaration,
+  EventDeclaration,
+  FunctionDeclaration,
+} from 'typechain'
+import { generateInputType, generateInputTypes } from './types'
+import { codegenFunctions } from './functions'
+
+export function codegenContractTypings(contract: Contract) {
+  const contractImports: string[] = ['Contract', 'ContractTransaction']
+  const allFunctions = values(contract.functions)
+    .map(codegenFunctions.bind(null, { returnResultObject: true }))
+    .join('')
+
+  if (allFunctions.match(/\W Overrides(\W|$)/)) contractImports.push('Overrides')
+  if (allFunctions.match(/\WPayableOverrides(\W|$)/)) contractImports.push('PayableOverrides')
+  if (allFunctions.match(/\WCallOverrides(\W|$)/)) contractImports.push('CallOverrides')
+
+  const template = `
+  import { ethers, EventFilter, Signer, BigNumber, BigNumberish, PopulatedTransaction } from 'ethers';
+  import { ${contractImports.join(', ')} } from '@ethersproject/contracts';
+  import { BytesLike } from '@ethersproject/bytes';
+  import { Listener, Provider } from '@ethersproject/providers';
+  import { FunctionFragment, EventFragment } from '@ethersproject/abi';
+
+  interface ${contract.name}Interface extends ethers.utils.Interface {
+    functions: {
+      ${values(contract.functions)
+        .map((v) => v[0])
+        .map(generateInterfaceFunctionDescription)
+        .join('\n')}
+    };
+
+    events: {
+      ${values(contract.events)
+        .map((v) => v[0])
+        .map(generateInterfaceEventDescription)
+        .join('\n')}
+    };
+  }
+
+  export class ${contract.name} extends Contract {
+    connect(signerOrProvider: Signer | Provider | string): this;
+    attach(addressOrName: string): this;
+    deployed(): Promise<this>;
+
+    on(event: EventFilter | string, listener: Listener): this;
+    once(event: EventFilter | string, listener: Listener): this;
+    addListener(eventName: EventFilter | string, listener: Listener): this;
+    removeAllListeners(eventName: EventFilter | string): this;
+    removeListener(eventName: any, listener: Listener): this;
+
+    interface: ${contract.name}Interface;
+
+    functions: {
+      ${values(contract.functions)
+        .map(codegenFunctions.bind(null, { returnResultObject: true }))
+        .join('\n')}
+    };
+
+    ${values(contract.functions).map(codegenFunctions.bind(null, {})).join('\n')}
```

Comment:
Does this (different options to `codegenFunctions` here) mean that in ethers v5 `contract.someMethod(...)` and `contract.functions.someMethod(...)` return a different type?

Reply:
Yes! I think that's not documented yet. But you can find it here https://github.com/ethers-io/ethers.js/issues/869